src/examples/common_plane_gpu.cpp

```diff
@@ -0,0 +1,280 @@
+  }
+  device.Print();
+
+  std::cout << "Creating MFEM mesh..." << std::endl;
+  axom::utilities::Timer timer(true);
+
+  mfem::Mesh mesh(mesh_file);
+
+  int ref_ct = 0;
+  for (int i{0}; i < ref_ct; ++i)
+  {
+    mesh.UniformRefinement();
+  }
+  std::cout << "MFEM mesh created (" << timer.elapsedTimeInMilliSec() << " ms)" << std::endl;
+
+  std::cout << "Creating MFEM grid function..." << std::endl;
+  timer.start();
+  mfem::H1_FECollection fe_coll(1, mesh.SpaceDimension());
+  mfem::FiniteElementSpace fe_space(&mesh, &fe_coll, mesh.SpaceDimension());
+  mfem::GridFunction coords(&fe_space);
+  mesh.SetNodalGridFunction(&coords, false);
+
+  auto coords_ptr = coords.Read();
+  auto x_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 0)];
+  auto y_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 1)];
+  auto z_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 2)];
+  std::cout << "MFEM grid function created (" << timer.elapsedTimeInMilliSec() << " ms)" << std::endl;
+
+  std::cout << "Creating Tribol connectivity..." << std::endl;
+  timer.start();
+
+  // mesh 1 connectivity (build on cpu)
+  auto mesh1_bdry_attrib = 4;
+  tribol::ArrayT<tribol::IndexT, 2, tribol::MemorySpace::Host> host_conn_1(num_contact_elems, 4);
+  int elem_ct = 0;
+  for (int be{0}; be < mesh.GetNBE(); ++be)
+  {
+    if (mesh.GetBdrAttribute(be) == mesh1_bdry_attrib)
+    {
+      mfem::Array<int> be_dofs(4);//, mfem::MemoryType::Host_UMPIRE);
+      fe_space.GetBdrElementDofs(be, be_dofs);
+      for (int i{0}; i < 4; ++i)
+      {
+        host_conn_1(elem_ct, i) = be_dofs[i];
+      }
+      ++elem_ct;
+    }
+  }
+  // move to gpu
+  tribol::ArrayT<tribol::IndexT, 2, MSPACE> conn_1(host_conn_1);
+
+  // mesh 2 connectivity (build on cpu)
+  auto mesh2_bdry_attrib = 5;
+  tribol::ArrayT<tribol::IndexT, 2, tribol::MemorySpace::Host> host_conn_2(num_contact_elems, 4);
+  elem_ct = 0;
+  for (int be{0}; be < mesh.GetNBE(); ++be)
+  {
+    if (mesh.GetBdrAttribute(be) == mesh2_bdry_attrib)
+    {
+      mfem::Array<int> be_dofs(4);//, mfem::MemoryType::Host_UMPIRE);
+      fe_space.GetBdrElementDofs(be, be_dofs);
+      for (int i{0}; i < 4; ++i)
+      {
+        host_conn_2(elem_ct, i) = be_dofs[i];
+      }
+      ++elem_ct;
+    }
+  }
+  // move to gpu
+  tribol::ArrayT<tribol::IndexT, 2, MSPACE> conn_2(host_conn_2);
+  std::cout << "Tribol connectivity created (" << timer.elapsedTimeInMilliSec() << " ms)" << std::endl;
+
+  std::cout << "Registering Tribol mesh data..." << std::endl;
+  timer.start();
+  constexpr tribol::IndexT mesh1_id = 0;
+  tribol::registerMesh(mesh1_id, num_contact_elems, fe_space.GetNDofs(), conn_1.data(), tribol::LINEAR_QUAD,
+    x_coords_ptr, y_coords_ptr, z_coords_ptr, MSPACE);
+  constexpr tribol::IndexT mesh2_id = 1;
+  tribol::registerMesh(mesh2_id, num_contact_elems, fe_space.GetNDofs(), conn_2.data(), tribol::LINEAR_QUAD,
+    x_coords_ptr, y_coords_ptr, z_coords_ptr, MSPACE);
+
+  constexpr tribol::RealT penalty = 5000.0;
```

Comment:
Minor: make a note that a constant penalty is used here (as opposed to the element thickness version).